[Approaches]: Raindrops (#339)
exercism/powershell: exercises/practice/raindrops/.approaches/switch-statement/content.md

# Using `switch` statement

```powershell
function Raindrops() {
    [CmdletBinding()]
    Param(
        [int]$Number
    )
    $sounds = switch (0) {
        ($Number % 3) {"Pling"}
        ($Number % 5) {"Plang"}
        ($Number % 7) {"Plong"}
        Default {$Number}
    }
    -join $sounds
}
```

This approach is similar to the `if` statements.
The value `0` is used to match if the input number is divisible for `3`, `5` and `7`.

> And if none of those give a match then the default will be the original number.

Reviewer: ```suggestion
And if none of those conditions are true, the default value will be the original number.
```
